src/tng-hooks.src.js

```diff
@@ -213,4 +224,45 @@
 			throw new Error("useEffect() only valid inside an Articulated Function or a Custom Hook.");
 		}
 	}
+
+	function useMemo(fn, ...guards) {
+		// passed in any guards?
+		if (guards.length > 0) {
+			// only passed a single guards list?
+			if (guards.length == 1 && Array.isArray(guards[0])) {
+				guards = guards[0];
+			}
+		}
+		// no guards passed
+		// NOTE: different handling than an empty guards list like []
+		else {
+			guards = undefined;
+		}
+
+		var bucket = getCurrentBucket();
+		if (bucket) {
+			// need to create this memoization-slot for this bucket?
+			if (!(bucket.nextMemoizationIdx in bucket.memoizations)) {
+				bucket.memoizations[bucket.nextMemoizationIdx] = [];
+			}
+
+			let memoizationIdx = bucket.nextMemoizationIdx;
```

Comment:
```suggestion
```

Reply:
We're not using an internal closure (like we are with `useEffect(..)`) so no need to create this variable for this hook.